torch/csrc/jit/runtime/operator.h

```diff
@@ -95,55 +112,73 @@ struct TORCH_API Operator {
       Symbol name,
       OperationCreator op_creator,
       c10::OperatorOptions options = c10::OperatorOptions())
-      : schema_(std::make_shared<FunctionSchema>(varArgSchemaWithName(name))),
-        op_creator_(std::move(op_creator)),
-        options_(std::move(options)) {}
+      : op_(c10::make_right<C10Operator, JitOnlyOperator>(JitOnlyOperator{
+          c10::make_left<FunctionSchema, std::string>(varArgSchemaWithName(name)),
+          std::move(options),
+          c10::make_right<Operation, OperationCreator>(std::move(op_creator))
+      })) {}
 
   Operation getOperation(const Node* node = nullptr) const {
-    if (op_) {
-      return *op_;
-    }
-    AT_ASSERT(node != nullptr);
-    return op_creator_(node);
+    return op_.fold<Operation>([] (const C10Operator& op) {
```

Comment:
I still find this lambda expressions hard to read. This code is not perf-critical (as it happens in initialization only) - why not just write an if statement? :)

Reply:
I think this is on the hot path for the jit compiler (i.e. relevant for compilation time).

I read those similar to how I read pattern matching statements in more descriptive languages, that makes it ok to read.